integration/internal/process_test.go

```diff
@@ -28,45 +37,143 @@ var _ = Describe("Start method", func() {
 		Consistently(processState.Session.ExitCode).Should(BeNumerically("==", -1))
 	})
 
-	Context("when process takes too long to start", func() {
-		It("returns a timeout error", func() {
-			processState := &ProcessState{}
-			processState.Path = "bash"
-			processState.Args = simpleBashScript
-			processState.StartTimeout = 200 * time.Millisecond
-			processState.StartMessage = "loop 5000"
-
-			err := processState.Start(nil, nil)
-			Expect(err).To(MatchError(ContainSubstring("timeout")))
+	Context("when a health check endpoint is provided", func() {
+		var server *ghttp.Server
+		BeforeEach(func() {
+			server = ghttp.NewServer()
 
-			Eventually(processState.Session.ExitCode).Should(Equal(143))
 		})
-	})
+		AfterEach(func() {
+			server.Close()
+		})
 
-	Context("when the command cannot be started", func() {
-		var (
-			processState *ProcessState
-		)
-		BeforeEach(func() {
-			processState = &ProcessState{}
-			processState.Path = "/nonexistent"
+		Context("when the healthcheck returns ok", func() {
+			BeforeEach(func() {
+				server.RouteToHandler("GET", "/healthz", ghttp.RespondWith(http.StatusOK, ""))
+			})
+
+			It("hits the endpoint, and successfully starts", func() {
+				processState.HealthCheckEndpoint = "/healthz"
+				processState.StartTimeout = 100 * time.Millisecond
+
+				url, err := url.Parse(server.URL())
+				Expect(err).NotTo(HaveOccurred())
+				processState.URL = *url
+
+				err = processState.Start(nil, nil)
+				Expect(err).NotTo(HaveOccurred())
+				Expect(server.ReceivedRequests()).To(HaveLen(1))
+				Consistently(processState.Session.ExitCode).Should(BeNumerically("==", -1))
+			})
 		})
 
-		It("propagates the error", func() {
-			err := processState.Start(nil, nil)
+		Context("when the healthcheck always returns failure", func() {
+			BeforeEach(func() {
+				server.RouteToHandler("GET", "/healthz", ghttp.RespondWith(http.StatusInternalServerError, ""))
+			})
+			It("returns a timeout error and stops health API checker", func() {
+				processState.HealthCheckEndpoint = "/healthz"
+				processState.StartTimeout = 500 * time.Millisecond
```

Comment:
(No action required) Consider using a variable or constant in place of a magic number where possible.  The name of a variable or constant can document the 'why' of a particular value, and either makes it easier to reuse a common value across tests.  For example:

```golang
shortTimeout = 100 * time.Millisecond
longTimeout = 500 * time.Millisecond
foreverTimeout = 20 * time.Second
```